node/chainSimulator/process/processor.go

```diff
@@ -0,0 +1,172 @@
+package process
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data"
+)
+
+type blocksCreator struct {
+	nodeHandler NodeHandler
+	blsKeyBytes []byte
+}
+
+func NewBlocksCreator(nodeHandler NodeHandler, blsKeyBytes []byte) (*blocksCreator, error) {
+	return &blocksCreator{
+		nodeHandler: nodeHandler,
+		blsKeyBytes: blsKeyBytes,
+	}, nil
+}
+
+// CreateNewBlock create and process a new block
+func (creator *blocksCreator) CreateNewBlock() error {
+	bp := creator.nodeHandler.GetProcessComponents().BlockProcessor()
+	currentHeader := creator.nodeHandler.GetChainHandler().GetCurrentBlockHeader()
+	var nonce, round uint64
+	var prevHash, prevRandSeed []byte
+	if currentHeader != nil {
+		nonce, round = currentHeader.GetNonce(), currentHeader.GetRound()
+		prevHash = creator.nodeHandler.GetChainHandler().GetCurrentBlockHeaderHash()
+		prevRandSeed = currentHeader.GetRandSeed()
+	} else {
+		prevHash = creator.nodeHandler.GetChainHandler().GetGenesisHeaderHash()
+		prevRandSeed = creator.nodeHandler.GetChainHandler().GetGenesisHeader().GetRandSeed()
+	}
+
+	newHeader, err := bp.CreateNewHeader(round+1, nonce+1)
+	if err != nil {
+		return err
+	}
+	err = newHeader.SetShardID(creator.nodeHandler.GetShardCoordinator().SelfId())
+	if err != nil {
+		return err
+	}
+
+	err = newHeader.SetPrevHash(prevHash)
+	if err != nil {
+		return err
+	}
+
+	err = newHeader.SetPrevRandSeed(prevRandSeed)
+	if err != nil {
+		return err
+	}
+
+	err = newHeader.SetPubKeysBitmap([]byte{1})
+	if err != nil {
+		return err
+	}
+
+	err = newHeader.SetChainID([]byte("chain"))
```

Comment:
"chain" should be a const and used also in node/chainSimulator/configs/configs.go L51

Reply:
done